lib/shttp/server.go

```diff
@@ -0,0 +1,125 @@
+package shttp
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"sync"
+
+	"github.com/lucas-clemente/quic-go/h2quic"
+	"github.com/netsec-ethz/scion-apps/lib/scionutil"
+	"github.com/scionproto/scion/go/lib/snet"
+)
+
+// Server wraps a h2quic.Server making it work with SCION
+type Server struct {
+	Addr string
+
+	s *h2quic.Server
+}
+
+var (
+	initOnce sync.Once
+	initErr  error
+)
+
+// ListenAndServeSCION listens for HTTPS connections on the SCION address addr and calls ServeSCION
+// with handler to handle requests
+func ListenAndServeSCION(addr, certFile, keyFile string, handler http.Handler) error {
+
+	laddr, err := snet.AddrFromString(addr)
+	if err != nil {
+		return err
+	}
+
+	// initialize SCION
+	initOnce.Do(func() {
+		if snet.DefNetwork == nil {
+			initErr = scionutil.InitSCION(laddr)
+		}
+	})
+	if initErr != nil {
+		return initErr
+	}
+
+	network := snet.DefNetwork
+	sconn, err := network.ListenSCION("udp4", laddr)
+	if err != nil {
+		return err
+	}
+
+	return ServeSCION(sconn, handler, certFile, keyFile)
```

Comment:
Why not just create a `Server` object here and call `Server::ListenAndServe` (just like in `ServeSCION` below)?